### PR TITLE
修改COPY-FROM No. 3 autograd

### PR DESCRIPTION
**Pull request by gouzil**
### PR types

Others
### PR changes

Others
### Description

修改COPY-FROM No. 3 autograd

- cn: PaddlePaddle/docs#5984

### CODE REVIEW

**Comment by gouzil** [comment]
@sunzhongkai588

File: python/paddle/autograd/py_layer.py
Comment (SigureMo) [inline python/paddle/autograd/py_layer.py:1]: 这个 PR 不要加 test=document_fix

Comment (SigureMo) [reply python/paddle/autograd/py_layer.py:1]: 应该是因为加了 test=document_fix 导致这边没有编译新的 wheel 包，进而 COPY 不过去，新提一个空的 commit 触发这边 build CI，之后重新触发 docs CI 试试

Comment (gouzil) [reply python/paddle/autograd/py_layer.py:1]: Done

Comment (gouzil) [reply python/paddle/autograd/py_layer.py:1]: merge 的应该也算吧

Comment (SigureMo) [reply python/paddle/autograd/py_layer.py:1]: 应该算吧

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM

**Review by luotao1 — COMMENTED** [review]
```
2023-07-03 23:49:22 ----------------End of the Check--------------------
2023-07-03 23:49:22 In addition, mistakes found in sample codes: samplecode_temp/paddle.autograd.PyLayer_example.py
2023-07-03 23:49:22 1 sample codes ran success
2023-07-03 23:49:22 1 sample codes ran failed
2023-07-03 23:49:22 samplecode_temp/paddle.autograd.PyLayer_example.py
2023-07-03 23:49:22 Mistakes found in sample codes. Please recheck the sample codes.'
2023-07-03 23:49:22 
```
示例代码单测失败

**Comment by sunzhongkai588** [comment]
> ```
> 2023-07-03 23:49:22 ----------------End of the Check--------------------
> 2023-07-03 23:49:22 In addition, mistakes found in sample codes: samplecode_temp/paddle.autograd.PyLayer_example.py
> 2023-07-03 23:49:22 1 sample codes ran success
> 2023-07-03 23:49:22 1 sample codes ran failed
> 2023-07-03 23:49:22 samplecode_temp/paddle.autograd.PyLayer_example.py
> 2023-07-03 23:49:22 Mistakes found in sample codes. Please recheck the sample codes.'
> 2023-07-03 23:49:22 
> ```
> 
> 
>      
>   
> 
> 示例代码单测失败

@wanghuancoder 辛苦review一下，因为代码是从docs下的中文文档原封不动迁移过来的。
PyLayerContext：  https://github.com/PaddlePaddle/docs/blob/develop/docs/api/paddle/autograd/PyLayerContext_cn.rst
PyLayer ：https://github.com/PaddlePaddle/docs/blob/develop/docs/api/paddle/autograd/PyLayer_cn.rst

File: python/paddle/autograd/py_layer.py [hunk 1]
```diff
@@ -266,6 +287,33 @@ class PyLayer(with_mateclass(PyLayerMeta, core.eager.PyLayer, PyLayerContext)):
...
+                @staticmethod
+                def backward(ctx, dy):
+                    # Get the tensors passed by forward.
+                    y, = ctx.saved_tensor()
+                    grad = dy * (1 - paddle.square(y))
+                    return grad
+
+            data = paddle.randn([2, 3], dtype="float64")
+            data.stop_gradient = False
+            z = cus_tanh.apply(data, func1=paddle.tanh)
```
Comment (wanghuancoder) [inline python/paddle/autograd/py_layer.py]: 示例报错，把如下代码删掉吧：
, func1=paddle.tanh

Comment (gouzil) [reply python/paddle/autograd/py_layer.py]: Done

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Pull request merged**